src/benchmark_runner.cc

```diff
@@ -258,19 +263,81 @@ bool BenchmarkRunner::ShouldReportIterationResults(
   // or because an error was reported.
   return i.results.has_error_ ||
          i.iters >= kMaxIterations ||  // Too many iterations already.
-         i.seconds >= min_time ||      // The elapsed time is large enough.
+         i.seconds >=
+             GetMinTimeToApply() ||  // The elapsed time is large enough.
          // CPU time is specified but the elapsed real time greatly exceeds
          // the minimum time.
          // Note that user provided timers are except from this test.
-         ((i.results.real_time_used >= 5 * min_time) && !b.use_manual_time());
+         ((i.results.real_time_used >= 5 * GetMinTimeToApply()) &&
+          !b.use_manual_time());
+}
+
+double BenchmarkRunner::GetMinTimeToApply() const {
+  // In order to re-use functionality to run and measure benchmarks for running
+  // a warmup phase of the benchmark, we need a way of telling whether to apply
+  // min_time or min_warmup_time. This function will figure out if we are in the
+  // warmup phase and therefore need to apply min_warmup_time or if we already
+  // in the benchmarking phase and min_time needs to be applied.
+  if (!warmup_done)
+    return min_warmup_time;
+  else
+    return min_time;
+}
+
+void BenchmarkRunner::FinishWarmUp(const IterationCount& i) {
+  warmup_done = true;
+  iters = i;
```

Comment:
i'm not sure why you're setting `iters` here if you then ignore it later.

Reply:
If I don't reset it here to the value it was before warmup started, the first run of the benchmarking phase will start with the last value of iters of the warmup phase, right?
The description of the issue explicitly stated, that the benchmarkings phase determination of a proper iters count should be unaffected by the warmup phase.

Reply:
ah yes, sorry. i missed that we were _resetting_ it and thought maybe we were setting it to what it was at the end of the warmup (the opposite of what your comment claims).